diff --git a/nexus/db-model/src/disk.rs b/nexus/db-model/src/disk.rs
index 4271e007b..00b68e9c0 100644
--- a/nexus/db-model/src/disk.rs
+++ b/nexus/db-model/src/disk.rs
@@ -79,6 +79,8 @@ pub struct Disk {
     /// For `Crucible` disks, see the DiskTypeCrucible model. For `LocalStorage`
     /// disks, see the DiskTypeLocalStorage model.
     pub disk_type: DiskType,
+
+    // XXX read-only flag
 }
 
 impl Disk {
diff --git a/nexus/src/app/disk.rs b/nexus/src/app/disk.rs
index 0b0038b59..41040799c 100644
--- a/nexus/src/app/disk.rs
+++ b/nexus/src/app/disk.rs
@@ -717,6 +717,7 @@ impl super::Nexus {
             project_id: authz_proj.id(),
             disk,
             snapshot_name: finalize_params.snapshot_name.clone(),
+            read_only: false, // XXX get this out of `disk.read_only()`
         };
 
         self.sagas
diff --git a/nexus/src/app/instance_platform/mod.rs b/nexus/src/app/instance_platform/mod.rs
index 6d077140a..f0978ecce 100644
--- a/nexus/src/app/instance_platform/mod.rs
+++ b/nexus/src/app/instance_platform/mod.rs
@@ -264,7 +264,7 @@ impl DisksByIdBuilder {
     ) -> Result<(), Error> {
         let backend =
             ComponentV0::CrucibleStorageBackend(CrucibleStorageBackend {
-                readonly: false,
+                readonly: false, // XXX eliza
                 request_json: volume.data().to_owned(),
             });
 
@@ -278,7 +278,7 @@ impl DisksByIdBuilder {
     ) -> Result<(), Error> {
         let backend = ComponentV0::FileStorageBackend(FileStorageBackend {
             path,
-            readonly: false,
+            readonly: false, // XXX eliza
             block_size: disk.block_size().to_bytes(),
             workers: Some(LOCAL_STORAGE_WORKERS),
         });
diff --git a/nexus/src/app/sagas/disk_create.rs b/nexus/src/app/sagas/disk_create.rs
index c971b60ff..0c6fa8fa9 100644
--- a/nexus/src/app/sagas/disk_create.rs
+++ b/nexus/src/app/sagas/disk_create.rs
@@ -658,6 +658,11 @@ async fn sdc_regions_ensure(
 
                 control: None,
 
+                // Hey you, don't set it just yet! Otherwise you can't import
+                // blocks into the disk!
+                //
+                // XXX when set to true, does this play well with region
+                // replacement?
                 read_only: false,
             },
         }],
diff --git a/nexus/src/app/sagas/finalize_disk.rs b/nexus/src/app/sagas/finalize_disk.rs
index 0a43a2016..e40ef74af 100644
--- a/nexus/src/app/sagas/finalize_disk.rs
+++ b/nexus/src/app/sagas/finalize_disk.rs
@@ -35,6 +35,7 @@ pub(crate) struct Params {
     pub project_id: Uuid,
     pub disk: datastore::CrucibleDisk,
     pub snapshot_name: Option<Name>,
+    pub read_only: bool,
 }
 
 declare_saga_actions! {
@@ -52,6 +53,9 @@ declare_saga_actions! {
     CLEAR_PANTRY_ADDRESS -> "clear_pantry_address" {
         + sfd_clear_pantry_address
     }
+    SET_READ_ONLY_IN_VCR -> "unused" {
+        + sfd_set_read_only_in_vcr
+    }
     SET_DETACHED_STATE -> "set_detached_state" {
         + sfd_set_detached_state
     }
@@ -127,6 +131,10 @@ impl NexusSaga for SagaFinalizeDisk {
 
         builder.append(clear_pantry_address_action());
 
+        if params.read_only {
+            builder.append(set_read_only_in_vcr_action());
+        }
+
         builder.append(set_detached_state_action());
 
         Ok(builder.build()?)
@@ -356,6 +364,14 @@ async fn sfd_clear_pantry_address(
     Ok(())
 }
 
+async fn sfd_set_read_only_in_vcr(
+    sagactx: NexusActionContext,
+) -> Result<(), ActionError> {
+    // blah blah some comment about setting this before setting state to
+    // Detached, because no instance can use it until then
+    todo!("change the VCR to set read_only = true")
+}
+
 async fn sfd_set_detached_state(
     sagactx: NexusActionContext,
 ) -> Result<(), ActionError> {
diff --git a/nexus/types/src/external_api/params.rs b/nexus/types/src/external_api/params.rs
index cbb58cf41..aad778f4f 100644
--- a/nexus/types/src/external_api/params.rs
+++ b/nexus/types/src/external_api/params.rs
@@ -2395,6 +2395,8 @@ pub struct DiskCreate {
 
     /// The total size of the Disk (in bytes)
     pub size: ByteCount,
+
+    pub read_only: bool,
 }
 
 // equivalent to crucible_pantry_client::types::ExpectedDigest
